### ISSUE TITLE
Single message to kafka not processed in some cases

### ISSUE THREAD

**Issue by neeleshs** [dibbhatt/kafka-spark-consumer#11]
I sent a single message to kafka topic with 3 partitions. It landed in partition 0. It looks like there is a chance in PartitionManager that _lastEnquedOffset == _lastComittedOffset, and with a single message in  _dataBuffer.  
```java
		if ((_lastEnquedOffset > _lastComittedOffset)
				&& (_waitingToEmit.isEmpty())) {

```
basically will wait for another message to come in to this partition before flushing out the last message. I had to change this condition to 
```java
		if ((_lastEnquedOffset >= _lastComittedOffset)
				&& (_waitingToEmit.isEmpty())) {

```
to avoid this issue.

**Comment by dibbhatt** [comment]
Cool. Thanks Neelesh. I will made this fix and checkin .
Dib 


     On Tuesday, 7 April 2015 10:38 AM, Neelesh <notifications@github.com> wrote:
   

 I sent a single message to kafka topic with 3 partitions. It landed in partition 0. It looks like there is a chance in PartitionManager that _lastEnquedOffset == _lastComittedOffset, and with a single message in _dataBuffer.         if ((_lastEnquedOffset > _lastComittedOffset)
                && (_waitingToEmit.isEmpty())) {
basically will wait for another message to come in to this partition before flushing out the last message. I had to change this condition to         if ((_lastEnquedOffset >= _lastComittedOffset)
                && (_waitingToEmit.isEmpty())) {
to avoid this issue.—

**Comment by dibbhatt** [comment]
I have made the changes you suggested . Thanks again to find this issue .

Dibyendu

**Comment by neeleshs** [comment]
Thanks for the quick response!
